packages/lit-html/src/lib/directives/live.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * @license
+ * Copyright (c) 2020 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {
+  directive,
+  Directive,
+  AttributePart,
+  noChange,
+  nothing,
+  PartInfo,
+  NODE_PART,
+  EVENT_PART,
+  PROPERTY_PART,
+  BOOLEAN_ATTRIBUTE_PART,
+  ATTRIBUTE_PART,
+} from '../lit-html.js';
+
+class LiveDirective extends Directive {
+  constructor(part: PartInfo) {
+    super();
+    if (part.type === EVENT_PART || part.type === NODE_PART) {
+      throw new Error(
+        'The `live` directive is not allowed on text or event bindings'
+      );
+    }
+    if (part.strings !== undefined) {
+      throw new Error('`live` bindings can only contain a single expression');
+    }
+  }
+
+  render(value: unknown) {
+    return value;
+  }
+
+  update(part: AttributePart, [value]: Parameters<this['render']>) {
+    const element = part.element;
+    const name = part.name;
+
+    // TODO (justinfagnani): This is essentially implementing a getLiveValue()
+    // method for each part type. Should that be moved into the AttributePart
+    // interface?
```

Comment:
Consider moving this to the `parts.ts` helpers?

Reply:
I don't think this is necessarily generally useful. The comment is more me thinking about polymorphism...